From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Milan Burda <milan.burda@gmail.com>
Date: Mon, 8 Apr 2019 20:08:20 +0200
Subject: Support ELECTRON_DEFAULT_ERROR_MODE in the GPU process


diff --git a/content/gpu/gpu_main.cc b/content/gpu/gpu_main.cc
index 39e87967360a9c7a2856ce4df9c182c782efb2a5..586babaff18c1d034a9b53d238164e390ae1dce2 100644
--- a/content/gpu/gpu_main.cc
+++ b/content/gpu/gpu_main.cc
@@ -9,6 +9,7 @@
 #include <utility>

 #include "base/bind.h"
+#include "base/environment.h"
 #include "base/lazy_instance.h"
 #include "base/message_loop/message_loop.h"
 #include "base/metrics/histogram_macros.h"
@@ -225,6 +226,10 @@ int GpuMain(const MainFunctionParams& parameters) {
       SEM_NOGPFAULTERRORBOX |
       SEM_NOOPENFILEERRORBOX);

+  auto env = base::Environment::Create();
+  if (env->HasVar("ELECTRON_DEFAULT_ERROR_MODE"))
+    SetErrorMode(GetErrorMode() & ~SEM_NOGPFAULTERRORBOX);
+
   // COM is used by some Windows Media Foundation calls made on this thread and
   // must be MTA so we don't have to worry about pumping messages to handle
   // COM callbacks.
